### PR TITLE
Attempt to make code nicer for FAQ

### PR DESCRIPTION
**Pull request by kozr**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

File had multiple components together and some of them could be reused by other templates so I refactored them out.

Also made some changes as I realized I was passing extra info to a simple rendering component

Side note, this was pretty good vvv
https://medium.com/hootsuite-engineering/react-re-render-optimization-7d369e0bf701

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Works on staging site

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/faqTemplates/ExpandableWithCategories.jsx [hunk 1]
```diff
@@ -52,60 +15,55 @@ const TitleImg = styled.img`
...
-const FaqSection = props => {
-  const [generalBin, setGeneralBin] = useState([])
-  const [logisticsBin, setLogisticsBin] = useState([])
-  const [teamsBin, setTeamsBin] = useState([])
+const ExpandableWithCategories = ({ faq, config }) => {
+  const [categorizedFaqMap, setFaqMap] = useState(new Map())
+
+  const categorizeFaq = faq => {
+    faq.forEach(({ category, question, answer }) => {
+      try {
```
Comment (ianmah) [inline src/components/faqTemplates/ExpandableWithCategories.jsx:23]: What kind of things are we trying to catch here?

File: src/components/faqTemplates/shared/ExpandableFaq.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React, { useState } from 'react'
```
Comment (ianmah) [inline src/components/faqTemplates/shared/ExpandableFaq.jsx:1]: Why the `shared` subfolder?

File: src/components/faqTemplates/ExpandableWithCategories.jsx [hunk 3]
```diff
@@ -52,60 +15,55 @@ const TitleImg = styled.img`
...
+        alt={config.titleAlt}
+        marginBottom={config.marginBottomTitle}
       />
-      <Spacers height={props.config.titleBottomSpacing} />
+      <Spacers height={config.titleBottomSpacing} />
       <Columns>
         <Column>
           General
-          <FaqList list={generalBin} />
+          <ExpandableFaqList list={categorizedFaqMap.get('General') ?? []} />
```
Comment (ianmah) [inline src/components/faqTemplates/ExpandableWithCategories.jsx:56]: wonder if the ExpandableFaqList  component should handle the empty case? Then we won't have to null check every time we use the component?

File: src/components/faqTemplates/ExpandableWithCategories.jsx (hunk 1 above)
Comment (kozr) [reply src/components/faqTemplates/ExpandableWithCategories.jsx:23]: Oh, I didn't commit my latest change which removed this block. I'll also remove the one from useEffect as well

File: src/components/faqTemplates/shared/ExpandableFaq.jsx (hunk 2 above)
Comment (kozr) [reply src/components/faqTemplates/shared/ExpandableFaq.jsx:1]: It's for components that may be useful for other templates under faq

Comment (kozr) [reply src/components/faqTemplates/shared/ExpandableFaq.jsx:1]: It was confusing to me to have all the files flat

File: src/components/faqTemplates/ExpandableWithCategories.jsx (hunk 3 above)
Comment (kozr) [reply src/components/faqTemplates/ExpandableWithCategories.jsx]: Sure, I'll pass the null check logic to `ExpandableFaqList`

File: src/components/faqTemplates/shared/FaqList.jsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import ExpandableFaq from './ExpandableFaq'
+
+const ExpandableFaqList = ({ list }) => {
+  return list
+    ? list.map(({ question, answer }) => (
```
Comment (ianmah) [inline src/components/faqTemplates/shared/FaqList.jsx:6]: does this work
```suggestion
  return list?.map(({ question, answer }) => (
```

**Review by ianmah — APPROVED** [review]

File: src/components/faqTemplates/shared/FaqList.jsx (hunk 4 above)
Comment (kozr) [reply src/components/faqTemplates/shared/FaqList.jsx:6]: ooo is this new in es2021?

Comment (kozr) [reply src/components/faqTemplates/shared/FaqList.jsx:6]: That makes it undefined, but I've put a `?? null` afterwards. I think it'll work.

**Review by ianmah — APPROVED** [review]

**Comment by ianmah** [comment]
:shipit:

**Pull request merged**